### ISSUE TITLE
Fix CustomResource in Python SDK

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-kubernetes#699]
This is a follow on issue from working on #435. Given the scope of the changes, I'm splitting this into a separate issue.

- [x] Update arg names to match the rest of the SDK (`__name__` -> `name`, `__opts__` -> `opts`)
- [x] Make `api_version` and `kind` required
- [x] Add `.get` method

**Comment by lukehoban** [comment]
> Add .get method
 
cc @cyrusnajmabadi who is working on this for tfgen Python right now as well.

**Comment by lukehoban** [comment]
Also cc https://github.com/pulumi/pulumi-kubernetes/issues/434 which is tracking get in python for kubernetes.

**Comment by joeduffy** [comment]
@lblackstone What is left in this work item? Judging by the summary description, we've already fixed the three sub-issues being tracked. Is it ready to be closed out, or there's more to do here?

**Comment by lukehoban** [comment]
I believe this is just waiting on #700 to merge.
